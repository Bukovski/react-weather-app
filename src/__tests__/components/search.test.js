import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import Search from "../../components/search.component";


describe("Search Component", () => {
	it('should render with valid data', () => {
		const { container } = render(
			<Search textValue={ "Text from value" }>
				Text Children
			</Search>
		);
		
		expect(screen.getByText('Text Children')).toBeInTheDocument();
		expect(screen.getByRole("search")).toHaveValue("Text from value");
		
		userEvent.click(screen.getByRole("button"));
		userEvent.type(screen.getByRole("search"), 'New');
		fireEvent.focusOut(screen.getByRole("search"));
		
		expect(container.firstChild).toMatchSnapshot();
	});
	
	it('click on button', () => {
		const testData = {
			textValue: "Text from value",
			onChangeValue: jest.fn(),
			onClearField: jest.fn(),
			onKeyCatcher: jest.fn(),
			onClickButton: jest.fn()
		}
		
		render(
			<Search
				onChangeValue={ testData.onChangeValue }
				onClearField={ testData.onClearField }
				onKeyCatcher={ testData.onKeyCatcher }
				onClickButton={ testData.onClickButton }
				textValue={ "Text from value" }
			>
				Text Children
			</Search>
		);
		
		expect(testData.onClickButton).toHaveBeenCalledTimes(0);
		expect(testData.onKeyCatcher).toHaveBeenCalledTimes(0);
		expect(testData.onChangeValue).toHaveBeenCalledTimes(0);
		expect(testData.onClearField).toHaveBeenCalledTimes(0);
		
		userEvent.click(screen.getByRole("button"));
		expect(testData.onClickButton).toHaveBeenCalledTimes(1);
		
		userEvent.type(screen.getByRole("search"), 'New');
		expect(testData.onKeyCatcher).toHaveBeenCalledTimes(3);
		expect(testData.onChangeValue).toHaveBeenCalledTimes(3);
		
		fireEvent.focusOut(screen.getByRole("search"))
		expect(testData.onClearField).toHaveBeenCalledTimes(1);
	});
});