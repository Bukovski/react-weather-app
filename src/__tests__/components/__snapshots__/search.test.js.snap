// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render with valid data 1`] = `
<div
  class="search"
>
  <div
    class="search__wrap"
  >
    <input
      class="search__input"
      placeholder="Search for a location"
      role="search"
      type="search"
      value="Text from value"
    />
    <button
      class="search__button"
      type="submit"
    >
      <svg
        class="search__icon"
        fill="none"
        height="24"
        width="24"
      >
        <path
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    Text Children
  </div>
</div>
`;
